api/src/main/java/jakarta/data/repository/Find.java

```diff
@@ -26,11 +26,29 @@
 
 
 /**
- * <p>Annotates a repository method returning entities as a parameter-based automatic query method.</p>
+ * <p>Annotates a repository method as a parameter-based automatic query method
+ * that returns entities or entity attributes.</p>
  *
- * <p>The {@code Find} annotation indicates that the annotated repository method executes a query to retrieve entities
- * based on its parameters and on the arguments assigned to its parameters. The method return type identifies the entity
- * type returned by the query. Each parameter of the annotated method must either:
+ * <p>The {@code Find} annotation indicates that the annotated repository
+ * method executes a query to retrieve entities (or a subset of entity attributes)
+ * based on its parameters and the arguments assigned to those parameters.</p>
+ *
+ * <p>The method return type identifies one of the following:</p>
+ * <ul>
+ * <li>the entity type returned by the query,</li>
+ * <li>a single entity attribute type returned by the query (requires the
+ *     {@link Select} annotation), or</li>
+ * <li>a Java record type representing a subset of entity attributes returned
+ *     by the query. The names of the record components and entity attributes
+ *     must match, or the entity attribute names must be specified by the
+ *     {@link Select} annotation.</li>
```

Comment:
Can we also use a regular class here?

Reply:
> Can we also use a regular class here?

With a little bit of added complexity, this approach could certainly be extended to include non-record classes as well.  However, I would propose that in this PR we stick with record only to start out with, get that working, and then separately consider if we want to add in non-record classes.  Would that be okay with you?

Reply:
I agree with Nathan.

Reply:
Baby-steps.
I like this approach.